# Парсинг - это хорошо

<p><b>Парсинг</b> - это автоматический сбор данных в интернете с использованием программных средств.</p>

<ul>
    <li><b>webbrowser</b> - это про открытие браузера на определенной странице;</li>
    <li><b>requests</b> - это для подключения к сайтам (программным способом);</li>
    <li><b>Beautiful Soup</b> - это для автоматического парсинга;</li>
    <li><b>selenium</b> - запускает браузер и управляет его работой.</li>
</ul>

In [1]:
import webbrowser

webbrowser.open('https://google.com/')

True

In [3]:
import requests

res = requests.get('https://google.com/fjkgjelr')

print(type(res))  # тип переменной - это ответ сервера к которому вы подключаетесь
print(res.status_code)  # статус подключения

<class 'requests.models.Response'>
404


## Коды ответов сервера
* **1xx** - подключение совершено, но запрос еще обрабатывается
* **2xx** -  подключение совершено успешно
* **3xx** - перенаправление
* **4хх** - ошибка в запросе - 404 страница не найдена
* **5хх** - 500, 501, 502 ошибка сервера. Запрос составлен ВЕРНО, но сервер не может его обработать.

## Запросы GET и POST

### GET

Когда мне необходимо ПОЛУЧИТЬ какую-то информацию от сервера, я использую **GET-запрос**.
Например: качаю картинку или музыку.

### POST 
Когда мне нужно отправить информацию серверу, я использую **POST**
Например: отсылаю форму заказа на сайте.

In [4]:
import requests

res = requests.get('https://gutenberg.org/cache/epub/1112/pg1112.txt')
res.status_code

200